docs/intro-react.md

```diff
@@ -0,0 +1,613 @@
+import { Text } from 'react-native';
+
+export default class MyPet extends Component {
+    render() {
+        return (
+            <Text>Hello, I am your pet!</Text>
+        );
+    }
+}
+```
+
+<block class="classical functional webNote androidNote" />
+
+Because JSX is JavaScript, you can use variables inside it. Here you are declaring a name for the pet, `petName`, and embedding it with curly braces inside `<Text>`.
+
+<div class="toggler">
+  <ul role="tablist" id="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTab('syntax', 'functional')">
+      Functional Syntax
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTab('syntax', 'classical')">
+      Class Syntax
+    </li>
+  </ul>
+</div>
+
+<block class="functional webNote androidNote" />
+
+```SnackPlayer name=Curly%20Braces
+import React from 'react';
+import { Text } from 'react-native';
+
+export default function MyPet() {
+  const petName = "Peebo";
+  return (
+    <Text>Hello, I am {petName}!</Text>
+  );
+}
+```
+
+<block class="classical webNote androidNote" />
+
+```SnackPlayer name=Curly%20Braces
+import React, { Component } from 'react';
+import { Text } from 'react-native';
+
+export default class MyPet extends Component {
+    render() {
+        const petName = "Peebo";
+
+        return (
+            <Text>Hello, I am {petName}!</Text>
+        );
+    }
+}
+```
+
+<block class="classical functional webNote androidNote" />
+
+## Custom Components
+
+You’ve already met [React Native’s Core Components](intro-react-native-components). React lets you nest these components inside each other to create new components. These nestable, reusable components are at the heart of the React paradigm.
+
+For example, you can nest [`Text`](text) and [`TextInput`](textinput) inside a [`View`](view) below, and React Native will render them together:
+
+<div class="toggler">
+  <ul role="tablist" id="toggle-syntax">
+    <li id="functional" class="button-functional" aria-selected="false" role="tab" tabindex="0" aria-controls="functionaltab" onclick="displayTab('syntax', 'functional')">
+      Functional Syntax
+    </li>
+    <li id="classical" class="button-classical" aria-selected="false" role="tab" tabindex="0" aria-controls="classicaltab" onclick="displayTab('syntax', 'classical')">
+      Class Syntax
+    </li>
+  </ul>
+</div>
+
+<block class="functional webNote androidNote" />
+
+```SnackPlayer name=Custom%20Components
+import React from 'react';
+import { Text, TextInput, View } from 'react-native';
+
+export default function MyPet() {
+    return (
+        <View>
+            <Text>Hello, I am...</Text>
+            <TextInput
+                style={{ height: 40, borderColor: 'gray', borderWidth: 1 }}
+                value="Name me!"
```

Comment:
This example has a hardcoded `TextInput` value, making it not editable. I don't think that was the intention?

Reply:
Should be placeholder, thanks!